### PR TITLE
[QUARKS-107] create SimulatedTemperatureSensor for recipes

### PR DESCRIPTION
**Pull request by dlaboss**
For the motivation / use of this see: https://github.com/apache/incubator-quarks-website/pull/33
Awaiting response to question in the pr.

### CODE REVIEW

File: samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * A Simulated temperature sensor.
+ * <p>
+ * The sensor starts off with an initial value.
+ * Each call to {@link #get()} changes the current value by
+ * a random amount between plus/minus a {@code deltaFactor}.
+ * The new current value is limited to a {@code maxTempRange}.
+ * <p>
+ * Temperature values are in Fahrenheit.
```
Comment (ddebrunner) [inline samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java]: 'Temperature values are in Fahrenheit."

Isn't that up to whoever is creating an instance?

The default constructor maps to a normal air temperature Fahrenheit range, but wouldn't something like this be suitable for Kelvin around the same range?

`new SimulatedTemperatureSensor(300, Ranges.closed(265, 315), 0.5);`

Comment (dlaboss) [inline samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java]: Good point.  Thx.

File: samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    private static final long serialVersionUID = 1L;
+    private static DecimalFormat df = new DecimalFormat("#.#");
+    private Random r = new Random();
+    private final Range<Double> maxTempRange;
+    private final double deltaFactor;
+    private double currentTemp;
+   
+    /**
+     * Create a temperature sensor.
+     * <p>
```
Comment (ddebrunner) [inline samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java]: It's a good habit in javadoc to close opened `<P>` with `</P>`.

Some HTML verification/sccessibility testing tools will complain about paragraphs that are not closed.

File: samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     *              Must be > 0.0
+     */
+    public SimulatedTemperatureSensor(double initialTemp,
+            Range<Double> maxTempRange, double deltaFactor) {
+        this.currentTemp = initialTemp;
+        this.maxTempRange = maxTempRange;
+        this.deltaFactor = deltaFactor;
+        Objects.requireNonNull(maxTempRange, "maxTempRange");
+        if (!maxTempRange.contains(currentTemp))
+            throw new IllegalArgumentException("currentTemp");
```
Comment (ddebrunner) [inline samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java]: The api argument is initialTemp.

File: samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        if (deltaFactor <= 0.0)
+            throw new IllegalArgumentException("deltaFactor");
+    }
+    
+    /** Get the maxTempRange setting */
+    public Range<Double> getMaxTempRange() {
```
Comment (ddebrunner) [inline samples/utils/src/main/java/quarks/samples/utils/sensor/SimulatedTemperatureSensor.java]: Why is this a "maximum temperature range"? Isn't it just a temperature range?

What would be a "minimum temperature range" ?

**Comment by dlaboss** [comment]
Unrelated WebClientSocketTest failure; likely load/timing related.

**Comment by dlaboss** [comment]
Same unrelated failure as noted above.  Will file a separate Jira.

**Pull request merged**